### ISSUE TITLE
Interactive highlights are not turned on using keyboard nav

### ISSUE THREAD

**Issue by Nancy-Salpepi** [phetsims/balloons-and-static-electricity#541]
**Test device**
MacBook Air (m1 chip) and iMac

**Operating System**
11.5.2 and 10.15.7

**Browser**
chrome and safari

**Problem description**
https://github.com/phetsims/qa/issues/706

When the interactive highlights toggle switch is moved to the ON position using keyboard nav, the interactive highlights do not work. 

**Steps to reproduce**
1. Enter the preferences menu using keyboard nav
2. In the Visual tab, turn on interactive highlights using keyboard nav
3. Exit the menu using escape or press the close (X) button. 
4. Switch to using the mouse--move it over the balloons, buttons, etc--no interactive highlights will appear. 

**Visuals**

https://user-images.githubusercontent.com/87318828/133313677-2d063c35-d64b-49a8-9b0a-8945ddb19d35.mov

Troubleshooting information:
!!!!! DO NOT EDIT !!!!!
Name: ‪Balloons and Static Electricity‬
URL: https://phet-dev.colorado.edu/html/balloons-and-static-electricity/1.5.0-rc.1/phet/balloons-and-static-electricity_all_phet.html
Version: 1.5.0-rc.1 2021-09-07 23:14:39 UTC
Features missing: applicationcache, applicationcache, touch
Flags: pixelRatioScaling
User Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/93.0.4577.63 Safari/537.36
Language: en-US
Window: 1351x659
Pixel Ratio: 2/1
WebGL: WebGL 1.0 (OpenGL ES 2.0 Chromium)
GLSL: WebGL GLSL ES 1.0 (OpenGL ES GLSL ES 1.0 Chromium)
Vendor: WebKit (WebKit WebGL)
Vertex: attribs: 16 varying: 31 uniform: 1024
Texture: size: 16384 imageUnits: 16 (vertex: 16, combined: 80)
Max viewport: 16384x16384
OES_texture_float: true
Dependencies JSON: {}

**Comment by jbphet** [comment]
During today's sound design meeting, we duplicated this, and noticed that it's because the preferences icon is still highlighted.  If the mouse is clicked anywhere, the highlight on the preferences menu goes away and interactive highlights start working.  We were wondering if the behavior could be changed such that if the user started moving the mouse - and perhaps moved it beyond some empirically determined threshold distance - the interactive highlights could come back into action.

**Comment by jbphet** [comment]
Marking as "blocks publication" during initial investigation.  We may change that if it turns out to be difficult to change.

**Comment by jessegreenberg** [comment]
> We were wondering if the behavior could be changed such that if the user started moving the mouse

This is how it used to work.  But the behavior shown in this issue is what I would expect now after changes @terracoda and I made for https://github.com/phetsims/balloons-and-static-electricity/issues/523. The decision is specifically in https://github.com/phetsims/balloons-and-static-electricity/issues/523#issuecomment-897034345.

Interactive Highlights will not activate while there is a highlight from DOM focus in the sim. Since interactive highlights look the same as focus highlights, it caused confusing cases when mouse movement or changes in the sim would deactivate the focus highlight. We decided that preventing interactive highlights while focus was in the sim was not ideal, but acceptable for now given time constraints. See examples in https://github.com/phetsims/balloons-and-static-electricity/issues/523 for examples of the confusing cases.

Assigning to @jbphet and @emily-phet to review or comment if we should still make other changes.

**Comment by emily-phet** [comment]
@jessegreenberg You may have already answered this, but what we were proposing was adding some logic such that the sim interprets that the mouse is now in use based on mouse movement, rather than mouse click. The result behavior would follow the same logic as currently, if focus highlight is present no interactive highlights are shown. The change would be that it would be focus is more readily removed once there is reasonable amount of mouse movement in the sim, rather than waiting for a mouse click to cause the transition between "end showing focus highlight" to "start showing interactive highlights".

I'm not sure that this was discussed over in BASE #523. If this is relatively straightforward to do, I think it would be desirable. If it's hard, then I think it's something we should consider for the future (and not block this publication).

**Comment by jessegreenberg** [comment]
Discussed during meeting:

I am going to implement the proposal in https://github.com/phetsims/balloons-and-static-electricity/issues/541#issuecomment-919451180 and https://github.com/phetsims/balloons-and-static-electricity/issues/541#issuecomment-920203746, where a certain amount of mouse movement will hide focus highlights and activate interactive highlights. Then, any keyboard press would re-activate focus highlights and hide interactive highlights.

**Comment by jbphet** [comment]
Yo @jessegreenberg - Do you have a sense of how hard it will be and how long it might take to implement the change you described in the previous comment?  We are looking to move forward on the 1.5 release of this sim, and need to decide whether to include this change.

**Comment by jessegreenberg** [comment]
I don't think it will be that difficult, I would suggest 2-4 hours of work. We discussed this in the Voicing check in meeting today and this should be higher on my priority list. I can work on this again on Friday and believe it will be by the end of 10/1/21.

**Comment by jbphet** [comment]
This issue was discussed in the 10/5/2021 sound design meeting, and we decided that if it isn't ready by midday this Friday, Oct 8, we will move forward with the next RC and will consider doing a maintenance release when the issue is resolved.

**Comment by jessegreenberg** [comment]
I am working on this in https://github.com/phetsims/joist/issues/751.

**Comment by Nancy-Salpepi** [comment]
Hey @jessegreenberg! I really like the switching back and forth between the interactive and focus highlights. I did notice this one case though on Mac + Chrome:

1. Turn on interactive highlights using the mouse. 
2. Move the mouse a tad so that the toggle button highlights. 
3. Tab to the close button and press space


![interactivehighlight](https://user-images.githubusercontent.com/87318828/137553341-1ea23f75-d08c-4b04-b696-ca9162072dbd.gif)
.

**Comment by Nancy-Salpepi** [comment]
@jessegreenberg I am seeing it happen whenever both types of focus are used in the preferences menu--not just in the instance I reported above.

<img width="776" alt="Screen Shot 2021-10-16 at 10 32 38 AM" src="https://user-images.githubusercontent.com/87318828/137591429-74880fcc-dd88-4de4-8083-6d8a99109403.png">

**Comment by Nancy-Salpepi** [comment]
@terracoda that was from the preferences dialog--the visual tab is highlighted. With Interactive highlights turned on, the mouse was hovering on that tab. Then I exited out of the dialog using the keyboard.

**Comment by jessegreenberg** [comment]
Linking against https://github.com/phetsims/qa/issues/721 so that it shows up as something to be patched in the next RC.

**Comment by jessegreenberg** [comment]
Change was pushed to a balloons-and-static-electricity-1.5 branch of scenery. With this change I was not able to get the problem to happen anymore. It seems pretty safe but I also tested the sim with Interactive Highlights for several minutes and all seemed OK to me.

**Comment by jessegreenberg** [comment]
dependencies.json has been updated for 1.5. Ready for verification in the next RC.

**Comment by Nancy-Salpepi** [comment]
Looks good.
